hardhat.config.js

```diff
@@ -69,11 +69,25 @@ task("upgrade-facets", "Upgrade existing facets, add new facets or remove existi
   .addParam("newVersion", "The version of the protocol to upgrade to")
   .addParam("env", "The deployment environment")
   .addParam("functionNamesToSelector", "JSON list of function names to selectors")
-  .addOptionalParam("facetConfig", "JSON list of facets to upgrade")
-  .setAction(async ({ env, facetConfig, newVersion, functionNamesToSelector }) => {
+  .addFlag("dryRun", "Test the upgrade without actually performing it")
+  .setAction(async ({ env, newVersion, functionNamesToSelector, dryRun }) => {
+    let balanceBefore, getBalance;
+    if (dryRun) {
+      let setupDryRun;
+      ({ setupDryRun, getBalance } = await lazyImport(`./scripts/util/dry-run.js`));
+      ({ env, deployerBalance: balanceBefore } = await setupDryRun(env));
+    }
+
     const { upgradeFacets } = await lazyImport("./scripts/upgrade-facets.js");
+    await upgradeFacets(env, null, newVersion, functionNamesToSelector);
```

Comment:
```suggestion
    await upgradeFacets(env, facetConfig, newVersion, functionNamesToSelector);
```
Related to the previous comment about `facetConfig` arg.